### ISSUE TITLE
DataFusion weekly project plan (Andrew Lamb) - Sep 16, 2024

### ISSUE THREAD

**Issue by alamb** [apache/datafusion#12494]
Follow on to https://github.com/apache/datafusion/issues/12391

My (personal) North ⭐ : 1000 projects are built using DataFusion 📈 ( Getting closer all the time )

**It would be great for other contributors to DataFusion to to make them visible somehow as well** 🙏  (feel free to copy / modify the format of this ticket)

## External news
(please feel free to post a comment to this ticket if you know of something else that should be highlighted)
* @emgeee and @ameyc talk about building streaming with DataFusion: https://techontherocks.show/3
* [Using StringView / German Style Strings to Make Queries Faster: Part 1 - Reading Parquet](https://www.influxdata.com/blog/faster-queries-with-stringview-part-one-influxdb/)
* [Using StringView / German Style Strings to Make Queries Faster: Part 2 - String Operations](https://www.influxdata.com/blog/faster-queries-with-stringview-part-two-influxdb/ 
* Cross post about StringView on DataFusion blog (waiting on review https://github.com/apache/datafusion-site/pull/25)

# Major Discussions (Please Take a Look)
* What features should DataFusion have? https://github.com/apache/datafusion/issues/12357

## Upcoming Meetups
* NYC tomorrow! https://github.com/apache/datafusion/discussions/11213
* Belgrade next week: https://github.com/apache/datafusion/discussions/11431
* San Francisco on Wednesday September 25th https://lu.ma/05lyf19h
* Seattle: Mon Oct 14: https://lu.ma/tnwl866b / https://github.com/apache/datafusion/discussions/10348#discussioncomment-10558368

## Highlights from last week
* We got a much needed upgrade in interval parsing correctness thanks to https://github.com/apache/datafusion/pull/12222
* @waruto210  and the team at Tencent started contributing: https://github.com/apache/datafusion/pull/12417 ❤ 
* @Rachelint  made MEDIAN and other non group aggregation faster for (really) high cardinality: https://github.com/apache/datafusion/pull/11827
* @andygrove  made the `42.0.0` release candidate and we are in the final phases of release: https://github.com/apache/datafusion/issues/11902
* Final touches on UF8View: https://github.com/apache/datafusion/pull/12309 from @devanbenz https://github.com/apache/datafusion/pull/12203 https://github.com/apache/datafusion/pull/12383 from @2010YOUY01 
* Started documenting extensions: https://github.com/apache/datafusion/pull/12420
* @Omega359 and @Abdullahsab3 supported `range` / `gen_series` for intervals less than `1 day` https://github.com/apache/datafusion/pull/12400
* @Omega359  supported https://github.com/apache/datafusion/pull/12283

## My (personal) plans for this week
- [ ] Figure out sqlparser donation https://github.com/sqlparser-rs/sqlparser-rs/issues/1294
- [x] Speak at NYC meetup and prepare for CMU building blocks seminar
- [ ] Get StringView enabled by default
- [x] Hack on `dft` with @matthewmturner : https://github.com/datafusion-contrib/datafusion-dft (it is a great testbed to see how easy/hard it is to keep [functions outside the core](https://github.com/apache/datafusion/issues/12357))
- [ ] Work down the review backlog

## Projects I plan to help actively help review / push forward
- [ ] StringView integration https://github.com/apache/datafusion/issues/11752
- [x] Parquet Filtering: https://github.com/apache/datafusion/issues/4028
- [ ] High Cardinality Grouping: https://github.com/apache/datafusion/issues/11679

# Algorithm for (my) prioritizing PR reviews
Note there are many [committers](https://projects.apache.org/committee.html?datafusion) who can and do review and merge PRs, so this is not the priorities of the project as a whole, just the approximate algorithm I use to prioritize my own time.

Priority:
1. Bug fixes (where something is just incorrect), especially regressions (where it used to work and now does not)
2. Improvements directly related to features needed for InfluxDB (my employer)
3. Documentation and test improvements (I view these as very strategically important)
4. PRs for other features I think are strategically important (See below)
5. Other new features / additions to functionality (note this is the lowest priority on purpose)

The top strategically important projects in my head are:
* Anything that improves DataFusion's quality such as bug fixes, and improved / expanded tests, etc
* Anything that makes it easier to use DataFusion as a user (docs, examples, better APIs, cleaner code, etc), including [Getting started guide for new users (who want to use DataFusion in their project) ](https://github.com/apache/arrow-datafusion/issues/7014) 
* Improved performance (with benchmarks)

Thus, if you are interested in contributing to DataFusion and are interested in a fast turn around time I would recommend looking into bug fixes / test improvements / documentation or the projects named above. 

If you propose adding new functionality, especially if the PR is large/complex and not connected to a wider need, the review cycle will likely be longer. You can make it a shorter cycle by looking at the comments on other recent PRs and following the same model (e.g. break it into small PRs, ensure there are tests in sqllogictest for example, the CI passes, includes documentation, etc)

## Background
The idea of this ticket is make my plans for DataFusion visible, largely for my own personal organizational needs, but also to:
1. Communicate / coordinate in the community 
2. Help provide an interesting summary of what is happening in DataFusion this week (both for my future self and others)

**Comment by alamb** [comment]
Review Queue

DataFusion (bugs / improvements)
- [ ] 

DataFusion (performance):
- [ ] https://github.com/apache/datafusion/pull/12395
- [ ] https://github.com/apache/datafusion/pull/12401
- [ ] https://github.com/apache/datafusion/pull/11943

DataFusion (new features, lower priority for me):
- [ ] https://github.com/apache/datafusion/pull/12186
- [ ] https://github.com/apache/datafusion/pull/12266
- [ ] https://github.com/apache/datafusion/pull/12225
- [ ] https://github.com/apache/datafusion/pull/12080
- [ ] https://github.com/apache/datafusion/pull/12295
- [ ] https://github.com/apache/datafusion/pull/11978

Arrow-rs
- [ ] https://github.com/apache/arrow-rs/pull/6376
- [ ] https://github.com/apache/arrow-rs/pull/6365
- [ ] https://github.com/apache/arrow-rs/pull/6290
- [ ] https://github.com/apache/arrow-rs/pull/6336


sqlparser-rs

**Comment by alamb** [comment]
For next week -- craeated https://github.com/datafusion-contrib/datafusion-functions-extra as a home for extra functions

See https://github.com/apache/datafusion/issues/12254#issuecomment-2356571864

**Comment by alamb** [comment]
Will try and revive these updates next week
